epictrack-web/src/components/myWorkplans/Filters/NameFilter.tsx

```diff
@@ -20,25 +28,64 @@ export const NameFilter = () => {
     );
   };
 
-  const debouncedSetSearchOptions = useRef(
-    debounce((searchText: string) => {
-      handleSearchOptions(searchText);
-    }, 1000)
-  ).current;
+  // Fetch project names from the backend when searchText changes
+  useEffect(() => {
+    const fetchProjectNames = async () => {

```

Comment:
Could we declare this outside of useffect?

Reply:
I do that usually, but this is actually the better pattern when dealing with an asynchronous effect. From chatgpt:
![image](https://github.com/bcgov/EPIC.track/assets/91914654/1164ac74-bd05-4aaa-828a-f1bbbb72b651)

Reply:
Interesting I didnt know that